unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/MapTestCase.java

```diff
@@ -20,15 +20,18 @@
 import static org.eclipse.collections.test.IterableTestCase.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
 import static org.junit.Assert.assertThrows;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 public interface MapTestCase
 {
-    <T> Map<Object, T> newWith(T... elements);
+    // Returns Object to allow subclasses to return either Map or MapIterable, and be cast to Map either way
+    <T> Object newWith(T... elements);
 
-    <K, V> Map<K, V> newWithKeysValues(Object... elements);
+    // Returns Object to allow subclasses to return either Map or MapIterable, and be cast to Map either way
+    <K, V> Object newWithKeysValues(Object... elements);
```

Comment:
Here's where we return Object to allow testing of Map and MapIterable together. The generics are funny here. Deleting them leads to the overriding methods not being recognized as overrides.